rasa/core/training/interactive.py

```diff
@@ -1724,7 +1724,7 @@ def calc_true_wrapping_width(text: Text, monospace_wrapping_width: int) -> int:
         if all(
             [
                 terminaltables.width_and_alignment.visible_width(line)
-                <= monospace_wrapping_width
+                <= monospace_wrapping_width + 1
```

Comment:
🤔 curious why this change is introduced

Reply:
Glad you asked this question, as I wasn't sure if this is the right approach.
This unit test was failing with a off by 1 error on the last test case:

```
FAILED tests/core/training/test_interactive.py::test_calc_true_wrapping_width - AssertionError: assert 59 == 60
 +  where 59 = <function calc_true_wrapping_width at 0x7efd101635b0>('牙龈出血的症状对应得疾病可能有：肥大性龈炎、骨髓增生异常综合征、口腔疾病、小儿出血性疾病、边缘性龈炎、获得性维生素K依赖性凝血因子异常、小儿白血病、回归热、坏死性龈口炎、青春期功能失调性子宫出血、郎-奥韦综合征、急性淋巴细胞白血病、感染性血小板减少性紫癜、单纯性牙周炎、单核细胞白血病、δ-贮存池病、小儿特发性血小板减少性紫癜、老年人真性红细胞增多症、急性根尖牙周炎、慢性根尖牙周炎、创伤性口炎、青少年牙周炎、慢性牙周炎', 119)
 +    where <function calc_true_wrapping_width at 0x7efd101635b0> = interactive.calc_true_wrapping_width
```
When I looked into the function internals more closely, I discovered it loops one level lower because in the [iterable](https://github.com/RasaHQ/rasa/blob/89671f0847009f6c2d34c09a919b53e6219b11fc/rasa/core/training/interactive.py#L1725) that gets used in `all()` function, there was an element (120) which fails the condition set by the test to be <= 119. So here I had the option a) change the unit test `wrapping_width` parameter, or b) modify the actual function. In hindsight, probably option a) would disturb the least the codebase. 
I couldn't  figure out why this test case only fails deterministically with Python 3.10, the libraries that get used in this function have not changed at all. Potentially something to do with the decoding of the text?

Reply:
I also don't know how the test parameters were chosen, perhaps @twerkmeister knows more as you seem to have committed this change?

Reply:
> Potentially something to do with the decoding of the text?

I would assume so, but can find any trace of it in the release note either. Sounds good

Reply:
Hmmm The context of this is the following ~ the rasa interactive console uses some table library that crashes when  text overflows. This does not happen when only dealing with latin alphabet. However many character sets in unicode such as Chinese characters are wider than a character in the latin alphabet when printed in console and the table layout does not account for that. That's why you need to calculate their width separately. 

What might have happened here is that the width of one or some of those characters changed in the unicode package of 3.10 🤔 making it wider and thus we have to wrap at 59 characters instead of 60 now for that specific string when the terminal width is 119. I think it's fine to increase the theoretical terminal width in the test to 120 and keep the target wrapping width at 60 if that then works in all supported python versions alike. Since the py310 change seems to make the result more conservative (wrapping already at 59 characters instead of 60) I'd say let's just roll with it and not worry too much.